--- httpclient.py.orig	2026-09-28 19:12:30.646397515 +0000
+++ httpclient.py	2026-09-28 19:12:30.657878498 +0000
@@ -19,6 +19,7 @@
 import ssl
 import time
 import urllib.parse
+import urllib.request
 from typing import AsyncIterator
 
 from . import signing as S
@@ -50,6 +51,12 @@
             if self.insecure:
                 ctx.check_hostname = False
                 ctx.verify_mode = ssl.CERT_NONE
+            proxy = urllib.request.getproxies().get("https")
+            if proxy and not urllib.request.proxy_bypass(self.host):
+                pu = urllib.parse.urlsplit(proxy)
+                c = http.client.HTTPSConnection(pu.hostname, pu.port or 80, timeout=timeout or self.timeout, context=ctx)
+                c.set_tunnel(self.host, self.port)
+                return c
             return http.client.HTTPSConnection(self.host, self.port, timeout=timeout or self.timeout, context=ctx)
         return http.client.HTTPConnection(self.host, self.port, timeout=timeout or self.timeout)
 
